### ISSUE TITLE
Deleting Files S3 Storage not working

### ISSUE THREAD

**Issue by JUVOJustin** [nextcloud/groupfolders#831]
Hi, i have used groupfolders for a while now. If i recall correctly, i have issues **deleting files only in groupfolders since ~ NC18**.

I use **s3 as primary storage**. Deleting files in users home directory seems to work finde.
Here is the corresponding log:

``` [remote] Error: TypeError: Argument 1 passed to OC\Files\Cache\CacheQueryBuilder::whereFileId() must be of the type integer, null given, called in /var/www/clients/client1/web89/web/nextcloud/lib/private/Files/Cache/Cache.php on line 655 at <<closure>>

 0. /var/www/clients/client1/web89/web/nextcloud/lib/private/Files/Cache/Cache.php line 655
    OC\Files\Cache\CacheQueryBuilder->whereFileId(null)
 1. /var/www/clients/client1/web89/web/nextcloud/lib/private/Files/Cache/Wrapper/CacheWrapper.php line 199
    OC\Files\Cache\Cache->moveFromCache(OCA\TermsOfServi ... {}, "__groupfolders/1/_DSC0502.jpg", "__groupfolders/ ... 1")
 2. /var/www/clients/client1/web89/web/nextcloud/apps/groupfolders/lib/Trash/TrashBackend.php line 150
    OC\Files\Cache\Wrapper\CacheWrapper->moveFromCache(OCA\TermsOfServi ... {}, "/__groupfolders/1/_DSC0502.jpg", "__groupfolders/ ... 1")
 3. /var/www/clients/client1/web89/web/nextcloud/apps/files_trashbin/lib/Trash/TrashManager.php line 105
    OCA\GroupFolders\Trash\TrashBackend->moveToTrash(OCA\Files_Trashb ... l}, "_DSC0502.jpg")
 4. /var/www/clients/client1/web89/web/nextcloud/apps/files_trashbin/lib/Storage.php line 193
    OCA\Files_Trashbin\Trash\TrashManager->moveToTrash(OCA\Files_Trashb ... l}, "_DSC0502.jpg")
 5. /var/www/clients/client1/web89/web/nextcloud/apps/files_trashbin/lib/Storage.php line 100
    OCA\Files_Trashbin\Storage->doDelete("_DSC0502.jpg", "unlink")
 6. /var/www/clients/client1/web89/web/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php line 266
    OCA\Files_Trashbin\Storage->unlink("_DSC0502.jpg")
 7. /var/www/clients/client1/web89/web/nextcloud/lib/private/Files/View.php line 1162
    OC\Files\Storage\Wrapper\Wrapper->unlink("_DSC0502.jpg")
 8. /var/www/clients/client1/web89/web/nextcloud/lib/private/Files/View.php line 716
    OC\Files\View->basicOperation("unlink", "/Projekte/_DSC0502.jpg", ["delete"])
 9. /var/www/clients/client1/web89/web/nextcloud/apps/dav/lib/Connector/Sabre/File.php line 448
    OC\Files\View->unlink("/Projekte/_DSC0502.jpg")
10. /var/www/clients/client1/web89/web/nextcloud/3rdparty/sabre/dav/lib/DAV/Tree.php line 179
    OCA\DAV\Connector\Sabre\File->delete()
11. /var/www/clients/client1/web89/web/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 287
    Sabre\DAV\Tree->delete("files/Admin/Projekte/_DSC0502.jpg")
12. <<closure>>
    Sabre\DAV\CorePlugin->httpDelete(Sabre\HTTP\Reque ... "}, Sabre\HTTP\Response {})
13. /var/www/clients/client1/web89/web/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php line 105
    call_user_func_array([Sabre\DAV\CorePlugin {},"httpDelete"], [Sabre\HTTP\Requ ... }])
14. /var/www/clients/client1/web89/web/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 479
    Sabre\Event\EventEmitter->emit("method:DELETE", [Sabre\HTTP\Requ ... }])
15. /var/www/clients/client1/web89/web/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 254
    Sabre\DAV\Server->invokeMethod(Sabre\HTTP\Reque ... "}, Sabre\HTTP\Response {})
16. /var/www/clients/client1/web89/web/nextcloud/apps/dav/lib/Server.php line 319
    Sabre\DAV\Server->exec()
17. /var/www/clients/client1/web89/web/nextcloud/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
18. /var/www/clients/client1/web89/web/nextcloud/remote.php line 165
    require_once("/var/www/client ... p")

DELETE /remote.php/dav/files/Admin/Projekte/_DSC0502.jpg
from 2.205.10.*** by Admin at 2020-03-24T19:26:51+00:00 ```

I am very sorry if i missed an issue template.

**Comment by mesiah1970-haydar** [comment]
same here

**Comment by stjosh** [comment]
Same for me. See also https://github.com/nextcloud/server/issues/19009 which seems to be very much related.

**Comment by tdsparrow** [comment]
Looks like it's an old issue with object storage, not only in NC18. 

When deleting a file in goupfolder, nc will move file then update database metainfo(L179-L181):
https://github.com/nextcloud/groupfolders/blob/e859eb0ce62379a6f2ebd7fef58a90d33771c22f/lib/Trash/TrashBackend.php#L167-L189

In turn it will call respective methods in object storage, both moveFromStorage and moveFromCache will update filecache table, the later one will failed to find the record of original file.

A naive fix could be adding check in groupfolder trashbin code.

**Comment by pierreozoux** [comment]
I thinks this is now corrected, in Nc 18.0.12, I can't reproduce the bug anymore (I did have this bug before).

Feel free to comment to reopen.
